### PR TITLE
test methods for product api

### PR DESCRIPTION
**Pull request by asurelaba**

### CODE REVIEW

File: src/main/java/com/solvd/carina/dummyjson/api/product/GetProductMethod.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+import com.zebrunner.carina.api.annotation.Endpoint;
+import com.zebrunner.carina.api.annotation.RequestTemplatePath;
+import com.zebrunner.carina.api.annotation.ResponseTemplatePath;
+import com.zebrunner.carina.api.annotation.SuccessfulHttpStatus;
+import com.zebrunner.carina.api.http.HttpMethodType;
+import com.zebrunner.carina.api.http.HttpResponseStatusType;
+import com.zebrunner.carina.utils.config.Configuration;
+
+
+@Endpoint(url = "${base_url}/product/1", methodType = HttpMethodType.GET)
```
Comment (asurelaba) [inline src/main/java/com/solvd/carina/dummyjson/api/product/GetProductMethod.java]: is this the right way to mention url for get by Id? May i know which is the best place to configure the id. I think it must be in test, if so, should i pass the id argument through constructor?

File: src/test/resources/api/products/_get/response.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  "price": "regex:[\\d]*",
+  "discountPercentage": "regex:[\\d]+.[\\d]*",
+  "rating": "regex:[\\d]+.[\\d]*",
+  "stock": "regex:[\\d]",
+  "brand": "type:String",
+  "category": "type:String",
+  "thumbnail": "type:String",
+  "images": [
+    "type:String"
+  ]
```
Comment (asurelaba) [inline src/test/resources/api/products/_get/response.json:14]: Please let me know if my understanding is correct. We use schema when we only want to validate structure and use response.json when we want to validate some values for the fields. So, in such case we could skip fields.

File: src/test/resources/api/products/_get/response.schema [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        },
+        {
+          "type": "string"
+        }
+      ]
+    }
+  },
+  "required": [
+    "id",
+    "title",
```
Comment (asurelaba) [inline src/test/resources/api/products/_get/response.schema:58]: Please let me know how to use schema when the api returns array.

File: src/main/java/com/solvd/carina/dummyjson/api/product/GetProductMethod.java (hunk 1 above)
Comment (ankochergin) [reply src/main/java/com/solvd/carina/dummyjson/api/product/GetProductMethod.java]: you can put id as a variable in constructor and use replaceUrlPlaceholder

File: src/test/java/com/solvd/carina/dummyjson/apitest/ProductTest.java [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+public class ProductTest implements IAbstractTest {
+
+    private static final Logger LOGGER = LogManager.getLogger(ProductTest.class);
+
+    @Test
+    @MethodOwner(owner = "asure")
+    public void testCreateProduct() {
+        LOGGER.info("testCreateProduct");
+
+        PostProductMethod api = new PostProductMethod();
```
Comment (ankochergin) [inline src/test/java/com/solvd/carina/dummyjson/apitest/ProductTest.java]:
```suggestion
    public void testCreateProduct() {
        PostProductMethod api = new PostProductMethod();
```

Comment (ankochergin) [inline src/test/java/com/solvd/carina/dummyjson/apitest/ProductTest.java]:
```suggestion
    public void testCreateProduct() {
        LOGGER.info("testCreateProduct");

        PostProductMethod postProductMethod = new PostProductMethod();
```

Comment (ankochergin) [reply src/test/java/com/solvd/carina/dummyjson/apitest/ProductTest.java]: please fix this in all tests

**Review by ankochergin — APPROVED** [review]
please add an xml suite and merge your PR

**Pull request merged**